### PR TITLE
Get rid of the dependency of Go compiler when WITH_GOLANG is OFF.

### PR DESCRIPTION
**Pull request by Xreki**
In CMake, [project](https://cmake.org/cmake/help/v3.0/command/project.html) is used to set a name, version, and enable languages for the entire project.

We can call [enable_language](https://cmake.org/cmake/help/v3.0/command/enable_language.html) to enable the `Go` language optionally.

It can fix problems in #8555 #5427 #3741 etc.

### CODE REVIEW

File: paddle/scripts/docker/README.md [hunk 1]
```diff
@@ -58,7 +58,7 @@ Users can specify the following Docker build arguments with either "ON" or "OFF"
 | `WITH_AVX` | OFF | Set to "ON" to enable AVX support. |
 | `WITH_TESTING` | OFF | Build unit tests binaries. |
 | `WITH_MKL` | ON | Build with [Intel® MKL](https://software.intel.com/en-us/mkl) and [Intel® MKL-DNN](https://github.com/01org/mkl-dnn) support. |
-| `WITH_GOLANG` | ON | Build fault-tolerant parameter server written in go. |
+| `WITH_GOLANG` | OFF | Build fault-tolerant parameter server written in go. |
```
Comment (typhoonzero) [inline paddle/scripts/docker/README.md]: May also change `paddle/scripts/docker/build.sh` to make this default OFF

Comment (Xreki) [reply paddle/scripts/docker/README.md]: https://github.com/PaddlePaddle/Paddle/blob/a67cebaf5f0c7657fc2f773360aeedb396a87d5f/paddle/scripts/docker/build.sh#L43

If changing the above value to `OFF`, then CI cannot test the go codes? Maybe I should change the value in `README.md` back to `ON`?

Comment (typhoonzero) [reply paddle/scripts/docker/README.md]: We can configure CI to enable golang explicitly to run the tests, the defaults can be still OFF.

Comment (Xreki) [reply paddle/scripts/docker/README.md]: Done.

**Pull request merged**